### PR TITLE
Untangle service from ovn requirement

### PR DESCRIPTION
**Pull request by booxter**
Note to reviewers: I advise to ignore white space changes when reviewing Functional test changes.

### CODE REVIEW

File: controllers/neutronapi_controller.go [hunk 1]
```diff
@@ -922,6 +923,14 @@ func (r *NeutronAPIReconciler) reconcileNormal(ctx context.Context, instance *ne
 		return ctrl.Result{}, err
 	}
 
+	if instance.IsOVNEnabled() {
+		err = r.reconcileExternalOVNSecrets(ctx, helper, instance, &secretVars)
```
Comment (booxter) [inline controllers/neutronapi_controller.go:927]: maybe reconcile should be executed anyway - to clean up secrets if the driver was disabled?

File: test/functional/neutronapi_controller_test.go [hunk 2]
```diff
@@ -1485,7 +1544,7 @@ var _ = Describe("NeutronAPI Webhook", func() {
 		)
 	})
 
-	When("A NeutronAPI instance is updated with unsupported fields", func() {
+	When("A NeutronAPI instance is configured", func() {
```
Comment (booxter) [inline test/functional/neutronapi_controller_test.go:1547]: should be reverted

File: api/bases/neutron.openstack.org_neutronapis.yaml [hunk 3]
```diff
@@ -1938,6 +1938,11 @@ spec:
                 default: memcached
                 description: Memcached instance name.
                 type: string
+              ml2Drivers:
+                description: 'TODO: describe the option'
```
Comment (booxter) [inline api/bases/neutron.openstack.org_neutronapis.yaml:1942]: should add a proper description

**Comment by booxter** [comment]
recheck

**Comment by booxter** [comment]
recheck

FIPS check failure, should be ok after https://github.com/openshift/release/pull/57159

**Comment by booxter** [comment]
`--- PASS: kuttl/harness/neutron_no_ovn (66.29s)` 🥳

**Comment by booxter** [comment]
@karelyatin @slawqo I think this should be ready for review now. Thanks!

File: api/bases/neutron.openstack.org_neutronapis.yaml [hunk 4]
```diff
@@ -2052,6 +2052,12 @@ spec:
                 default: memcached
                 description: Memcached instance name.
                 type: string
+              ml2Drivers:
```
Comment (slawqo) [inline api/bases/neutron.openstack.org_neutronapis.yaml:2055]: shouldn't we name it like e.g. "ml2MechanismDrivers" as there are different types of drivers in ML2 plugin (type drivers, mechanism drivers, extension drivers for example)

**Comment by booxter** [comment]
@karelyatin do you have any input here before I respin for what Slawek asked? (Renaming the field.)

File: api/v1beta1/neutronapi_types.go [hunk 5]
```diff
@@ -102,6 +102,10 @@ type NeutronAPISpecCore struct {
 	// PreserveJobs - do not delete jobs after they finished e.g. to check logs
 	PreserveJobs bool `json:"preserveJobs"`
 
+	// +kubebuilder:validation:Optional
+	// Ml2Drivers - list of ml2 drivers to enable. Using ["ovn"] if not set.
```
Comment (karelyatin) [inline api/v1beta1/neutronapi_types.go:106]: wondering if can't we set default here itself or as part of webook instead of setting in GetMl2Drivers?

**Comment by booxter** [comment]
@karelyatin @slawqo In the latest update, I:

- renamed the ml2Drivers field into ml2MechanismDrivers;
- implemented defaulting using CRD kubebuilder to fall back to ovn.

**Pull request merged**